internal/testrunner/runners/policy/policy.go

```diff
@@ -184,6 +214,46 @@ func cleanPolicy(policy []byte, entriesToClean []policyEntryFilter) ([]byte, err
 	return yaml.Marshal(policyMap)
 }
 
+// replaceOtelComponentIDs finds OTel Collector component IDs in the policy and replaces them with controlled values.
+// It also replaces references to those IDs in service.extensions and service.pipelines.
+func replaceOtelComponentIDs(policy []byte) []byte {
+	replacementsDone := map[string]string{}
+
+	policy = otelComponentIDsRegexp.ReplaceAllFunc(policy, func(match []byte) []byte {
+		count := 0
+		lines := bytes.Split(match, []byte("\n"))
+		for i, line := range lines {
+			line = bytes.TrimRight(line, "\r\n")
+			stringLine := string(line)
```

Comment:
Nit. Instead of keeping an array with the lines that we need to join later, we could use an `Scanner` to read the lines, and a `strings.Builder` to build the result. In principle the `Scanner` would take care of the newline characters and so on.

Reply:
That approach looks better, thanks!
Updated code in https://github.com/elastic/elastic-package/pull/2873/commits/0c5483c9ba7c95b7d7bc5b0f147ba8dfa65ad326